superset/models/helpers.py

```diff
@@ -425,7 +425,19 @@ def changed_on_utc(self) -> str:
 
     @property
     def changed_on_humanized(self) -> str:
-        return humanize.naturaltime(datetime.now() - self.changed_on)
+        try:
+            from superset import conf
+            from flask_babel import get_locale
+            langs = conf.get("LANGUAGES", None)
+            locale = str(get_locale())
+
+            if "_" in locale:
+                humanize.i18n.activate(locale)
+            elif langs is not None:
+                humanize.i18n.activate("{}_{}".format(locale,
+                                                      langs[locale]["flag"].upper()))
```

Comment:
I'd assume this locale setting is global. You might want to call

```python
humanize.i18n.deactivate()
```

to avoid localization being polluted across different request context.

Maybe create a helper function in `superset/utils/i18n.py`?

```python
def localized_naturaltime(time: datetime, locale: str) -> str:
   humanize.i18n.activate(locale)
   naturalized = humanize.naturaltime(time)
   humanize.i18n.deactivate(locale)
   return naturalized
```

Reply:
@ktmud
I've searched in superset, there is only one place where humanize is used, so that I think we may need not call the activate method.
I've tested most pages and found no problems (including switching languages).

Reply:
I once added the code to change the locale after flask was initialized, but it didn't seem to take effect at that time. I'll do another verification later to see if I remember wrong.

Reply:
It's only used here doesn't mean it will not be used elsewhere in the future. The pollution can happen when someone with another language came in and visit a page that used `humanize` without `activate` being called, they may see the global locale set by another use in another procedure, I.e.

```
def func1:
   humanize.activate(...)

def func2:
   no humanize activate

user 1 (locale = zh) -> func1 -> zh
user 2 (local = fr) -> func2 -> zh
```

So it's safer if you always deactivate.